com.unity.ml-agents/Runtime/Communicator/RpcCommunicator.cs

```diff
@@ -455,8 +456,31 @@ UnityInputProto Exchange(UnityOutputProto unityOutput)
                 QuitCommandReceived?.Invoke();
                 return message.UnityInput;
             }
+            catch (RpcException rpcException)
+            {
+                // Log more verbose errors if they're something the user can possibly do something about.
+                switch (rpcException.Status.StatusCode)
+                {
+                    case StatusCode.Unavailable:
+                        // This can happen when python disconnects. Ignore it to avoid noisy logs.
+                        break;
+                    case StatusCode.ResourceExhausted:
+                        // This happens is the message body is too large. There's no way to
+                        // gracefully handle this, but at least we can show the message and the
+                        // user can try to reduce the number of agents or observation sizes.
+                        Debug.LogError($"GRPC Exception: {rpcException.Message}. Exiting communicator.");
+                        break;
+                    default:
+                        // Other unknown errors. Keep these quiet for now.
+                        break;
+                }
+                m_IsOpen = false;
+                QuitCommandReceived?.Invoke();
+                return null;
+            }
             catch
             {
+                // Fall-through for other error types
```

Comment:
Same comment about logging here - I don't think we ever hit it, so it would be nice to know when we do, but it's also potentially noisy to users.